bob-apps/bin/config_cluster_generator/pattern.rs

```diff
@@ -0,0 +1,267 @@
+use anyhow::{anyhow, Result as AnyResult};
+use bob_common::configs::cluster::Node;
+use bob_common::core_types::{DiskName, DiskPath};
+use itertools::Itertools;
+
+fn parse_address_pattern(pattern: &String) -> AnyResult<((String, u16), String)> {
+    debug!("Pattern to parse: {}", pattern);
+    let re: regex::Regex = regex::Regex::new(r"^([\w.]+):(\d+)(/[\w/]+)$").unwrap();
+
+    if let Some(captures) = re.captures(pattern) {
+        let ip = captures.get(1).unwrap().as_str().to_owned();
+        let port = captures.get(2).unwrap().as_str();
+        let path = captures.get(3).unwrap().as_str().to_owned();
+
+        let port: u16 = port
+            .parse()
+            .map_err(|_| anyhow!("Failed to parse port {}", port))?;
+        Ok(((ip, port), path))
+    } else {
+        Err(anyhow!("Failed to parse the pattern {}", pattern))
+    }
+}
+
+fn substitute_node(node_pattern: &str, ip: &str, port: u16, id: usize) -> String {
+    node_pattern
+        .replace("{ip}", &ip)
+        .replace("{port}", &port.to_string())
+        .replace("{id}", &id.to_string())
+}
+
+fn generate_range_samples(pattern: &str) -> impl Iterator<Item = String> {
+    let re = regex::Regex::new(r"\[(\d+)-(\d+)]").unwrap();
+
+    let ranges = re.captures_iter(pattern).map(|captures| {
+        let start: usize = captures[1].parse().unwrap();
+        let end: usize = captures[2].parse().unwrap();
+        start..=end
+    });
+
+    re.split(pattern)
+        .zip_longest(ranges)
+        .map(|x| {
+            if let itertools::EitherOrBoth::Both(part, range) = x {
+                range.map(|i| part.to_string() + &i.to_string()).collect()
+            } else {
+                vec![x.left().unwrap().to_string()]
```

Comment:
IMO it's better to use `expect` with some clarification why we don't expect error here, like `split > ranges`